packages/contracts/contracts/maci/MACI.sol

```diff
@@ -51,16 +51,26 @@ contract MACI is Ownable, BaseMACI, ICommon {
 
   /// @notice Initialize the poll by given poll id and transfer poll ownership to the caller.
   /// @param pollId The poll id
-  function initPoll(uint256 pollId, address registryAddress) public onlyOwner {
+  function initPoll(uint256 pollId) public onlyOwner {
     PollContracts memory pollAddresses = polls[pollId];
     IPoll poll = IPoll(pollAddresses.poll);
 
-    poll.setRegistry(registryAddress);
     poll.init();
     poll.transferOwnership(msg.sender);
   }
 
   /// @notice Set RegistryManager for MACI
+  /// @param pollId The poll id
+  /// @param registryAddress The registry address
+  function setPollRegistry(uint256 pollId, address registryAddress) public onlyOwner {
```

Comment:
When I added subgraph, I figured out that I need to link poll with registry because registry should be available before poll initialization. That's why I added it as a separate step. I don't think it will be possible to reuse `deployPoll` and set registry for the poll because registry address is not available there.
Transferring ownership won't work as well because `deployPoll` caller is `MACI`.